04.ls/ls.rb

```diff
@@ -1,22 +1,39 @@
 #! /usr/bin/env ruby
 # frozen_string_literal: true
 
+require 'optparse'
+
 COLUMN_SIZE = 3
 PADDING_MARGIN = 1
 
-items = Dir.children(__dir__)
-sorted_items = items.sort.reject { |item| item.start_with?('.') }
+argv_options = ARGV.getopts('a')
+
+def main(options = {})
+  file_names = file_names(options)
+
+  padding_size = file_names.map(&:length).max + PADDING_MARGIN
 
-padding_size = sorted_items.map(&:length).max + PADDING_MARGIN
+  lack_count = file_names.length % COLUMN_SIZE
+  file_names.concat([''] * (COLUMN_SIZE - lack_count)) if lack_count.positive?
 
-lack_count = sorted_items.length % COLUMN_SIZE
-sorted_items.concat([''] * (COLUMN_SIZE - lack_count)) if lack_count.positive?
+  row_size = (file_names.length / COLUMN_SIZE)
+  matrix = file_names.each_slice(row_size).to_a.transpose
+  matrix.each do |row_file_names|
+    row_file_names.each do |file_name|
+      print file_name.ljust(padding_size)
+    end
+    puts
+  end
+end
 
-row_size = (sorted_items.length / COLUMN_SIZE)
-matrix = sorted_items.each_slice(row_size).to_a.transpose
-matrix.each do |row_items|
-  row_items.each do |item|
-    print item.ljust(padding_size)
+def file_names(options = {})
+  file_names = Dir.entries(__dir__)
+  sorted_file_names = file_names.sort
```

Comment:
```suggestion
  file_names = Dir.entries(__dir__).sort
```

変数代入の時点でsortしてしまっていいのでは？